paasta_tools/metrics/metastatus_lib.py

```diff
@@ -259,37 +258,45 @@ def assert_tasks_running(metrics):
     )
 
 
-def assert_no_duplicate_frameworks(state, frameworks_list):
-    """A function which asserts that there are no duplicate frameworks running, where
-    frameworks are identified by their name.
-
-    Note the extra spaces in the output strings: this is to account for the extra indentation
-    we add, so we can have:
-
-        frameworks:
-          framework: marathon count: 1
+def assert_framework_count(state, configured_framework_count):
```

Comment:
this seems more complicated than it needs to be. why're we making one 'mega' check rather than the function per check that we had before, allowing the caller to filter the frameworks checked like we already did in ``assert_no_duplicate_frameworks`` ? I think you could've just updated the counting code that was on line 227 to match whether the framework name startswith the same value in ``frameworks_list`` rather than the exact match?

Reply:
@Rob-Johnson, this might look like I'm making a mega-check, but I'm not in fact.
I'm just removing several unused or broken assertions.

assert_active_frameworks() - is not used at all.

assert_connected_frameworks, assert_disconnected_frameworks, assert_inactive_frameworks are called by get_framework_metrics_status which is, in its turn, called by paasta_metastatus.
There 3 assertions relies on mesos counters that only provide number, but don't specify which framework is connected, inactive or disconnected.

```
  "master/frameworks_active": 3,
  "master/frameworks_connected": 3,
  "master/frameworks_inactive": 1,
  "master/frameworks_disconnected": 1,
```

It was possible to check these number before we started using task processing frameworks because we knew how many frameworks we were supposed to have connected. But these days we might have 1 disconnected marathon framework and 1 connected task proc framework and assert_connected_frameworks wouldn't notice any problem. These numbers are not longer useful in my opinion.

As for frameworks_list we have in assert_no_duplicate_frameworks, it is hardcoded at [metastatus_lib.py](https://github.com/Yelp/paasta/blob/master/paasta_tools/metrics/metastatus_lib.py#L584).
The only place where the caller can actually set this value is [check_mesos_duplicate_frameworks.py](https://github.com/Yelp/paasta/blob/master/paasta_tools/monitoring/check_mesos_duplicate_frameworks.py#L44).
In other words if we need to add a new framework name to the list, we'll have to update it in puppet and update it one more time in paasta_tools.

I think I disagree that this is actually better than hardcoding it in the function where is it used indeed so that we need to update it in paasta_tools only.

It would be great if we had framework names in marathon.json. In this case we wouldn't need to maintain frameworks_list.